### PR TITLE
Embeddables object as query parameter

### PR DESCRIPTION
**Pull request by mishal**

### CODE REVIEW

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 1]
```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
...
+    /**
+     * @group 5936
+     */
+    public function testProcessParameterValueEmbeddable()
+    {
+        $query  = $this->_em->createQuery("SELECT a FROM " . DDC3597Image::class . " a");
+
+        // load class meta data
+        $this->_em->getClassMetadata(DDC3597Image::class);
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Image::class));
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Assertion not needed

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 2]
```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
...
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Dimension::class));
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Assertion not needed

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 3]
```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
...
 
+    /**
+     * @group 5936
+     */
+    public function testProcessParameterValueEmbeddable()
+    {
+        $query  = $this->_em->createQuery("SELECT a FROM " . DDC3597Image::class . " a");
+
+        // load class meta data
+        $this->_em->getClassMetadata(DDC3597Image::class);
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: The fact that you need to manually trigger metadata loading is a smell

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 4]
```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
...
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Image::class));
+        $this->assertTrue($this->_em->getMetadataFactory()->hasMetadataFor(DDC3597Dimension::class));
+
+        $this->assertInstanceOf(
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Can you a

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 5]
```diff
@@ -183,6 +186,24 @@ public function testProcessParameterValueClassMetadata()
...
+            DDC3597Dimension::class,
+            $query->processParameterValue(new DDC3597Dimension())
+        );
```
Comment (Ocramius) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Could you please assert on the SQL string and the bound parameters?

Comment (mishal) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: Yes, but testing bound parameters will not be enough. the problems are when executing the query, it calls processParameterValue() which is the problematic part.

Comment (mishal) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php]: I based the test on this one: https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Query/QueryTest.php#L177

**Pull request closed without merging**